utils/zerovec/src/ule/mod.rs

```diff
@@ -189,6 +189,9 @@ pub trait AsVarULE {
 ///
 /// # Safety
 ///
+/// There must be no padding bytes involved in this type: [`Self::as_byte_slice()`] MUST return
+/// a slice of initialized bytes provided that `Self` is initialized.
```

Comment:
1. The safety docs are partly here and partly on the `from_byte_slice_unchecked` function
2. Can we / should we say something like, `mem::size_of_val(bytes)` must equal `mem::size_of_val(self)` ?

Reply:
1. Done. I wrote new docs instead of moving them, `from_byte_slice_unchecked` still has duplicated safety docs, but I think it's good to be redundant for unsafe stuff.
2. I don't think we need to, everything is totally safe even if the bytes are a subset, it just becomes nearly impossible to implement `parse_byte_slice` correctly.

Reply:
I was thinking about equality when I said that.

Reply:
https://github.com/unicode-org/icu4x/issues/1083